static/templates/compose.hbs

```diff
@@ -65,17 +65,16 @@
                             <template id="compose_close_and_save_tooltip">{{t 'Cancel compose and save draft' }} <span class="hotkey-hint">(Esc)</span></template>
                         </div>
                         <div id="stream-message" class="order-1">
-                            <div class="stream-selection-header-colorblock message_header_stream left_part" tab-index="-1"></div>
                             <div class="right_part">
-                                <span id="compose-lock-icon">
-                                    <i class="fa fa-lock" title="{{t 'This is a private stream' }}" aria-hidden="true"></i>
-                                </span>
-                                <span id="compose-globe-icon">
-                                    <i class="zulip-icon zulip-icon-globe" title="{{t 'This is a web-public stream' }}" aria-hidden="true"></i>
-                                </span>
                                 <a role="button" class="narrow_to_compose_recipients zulip-icon zulip-icon-arrow-left-circle order-1" data-tooltip-template-id="narrow_to_compose_recipients_tooltip" tabindex="0">
                                 </a>
-                                <input type="text" class="recipient_box" name="stream_message_recipient_stream" id="stream_message_recipient_stream" maxlength="{{ max_stream_name_length }}" value="" placeholder="{{t 'Stream' }}" autocomplete="off" tabindex="0" aria-label="{{t 'Stream' }}" />
+                                <div class="stream_selection_dropdown new-style">
```

Comment:
What's the thinking on the name `stream_selection_dropdown`? That feels like a good generic name for a reusable "select a stream" component, but this class doesn't apply to similar components in the "move topic" UI, for example. I feel like it'd be nice to have `compose` in the name of the main selector for a component that's only used in the compose box, and I would consider using an ID rather than a class for a component that is conceptually a singleton. (I kinda expect this one will be, because a lot of the selector logic appears related to specific compose box UI details that may not be relevant for similar dropdowns elsewhere).

Also, is the `new-style` CSS class required? If it is, we might want to do a mini prep project to fix that by moving CSS -- I don't think there is a "non-new-style" variant for any of the components inside this div. 

(I think roughly it'll involve making CSS for these components that's currently inside a `new-style` block to just be the CSS for that block, but haven't looked)

Reply:
Great point; I've updated the classname.

> is the new-style CSS class required?

without it, the dropdown looks like this:

<img width="148" alt="image" src="https://user-images.githubusercontent.com/5634097/213846339-16be1c63-d7f6-4a5a-a384-96c9c704de05.png">

> If it is, we might want to do a mini prep project to fix that by moving CSS -- I don't think there is a "non-new-style" variant for any of the components inside this div.
>
> (I think roughly it'll involve making CSS for these components that's currently inside a new-style block to just be the CSS for that block, but haven't looked)

Can you explain a bit further what you mean by '"non-new-style" variant' and why that's relevant? I think I'm missing some context on what we use `new_style` for.

Reply:
Let me start a #frontend thread about this.

Reply:
https://chat.zulip.org/#narrow/stream/6-frontend/topic/new-style.20CSS.20class/near/1498423

Reply:
For this PR, we can probably just leave `new-style` and hope to remove it in the sweeps described there.